lib/src/io.cpp

```diff
@@ -466,7 +467,18 @@ noexcept (false) {
         dl::obname tmp{ dl::origin{ origin },
                         dl::ushort{ copy },
                         dl::ident{ std::string{ id, id + idlen } } };
-        xs.emplace_back(tmp.fingerprint("FRAME"), tell);
+
+        auto fp = tmp.fingerprint("FRAME");
+        /* Although index.size() often is fairly large, the number of
+         * unique frames are typically just an handful. Hence the repeated
+         * calls to find() are not so bad as it might seem at first glace.
+         */
+        const auto itr = xs.find(fp);
+        if (itr == xs.end()) {
+            xs.insert( { std::move(fp), { tell } });
+        } else {
+            itr->second.push_back( tell );
+        }
```

Comment:
You are right, and that should be fine:

> Reference to the mapped value of the new element if no element with key key existed. Otherwise a reference to the mapped value of the existing element whose key is equivalent to key.

https://en.cppreference.com/w/cpp/container/map/operator_at